.evergreen/csfle/kms_failpoint_server.py

```diff
@@ -0,0 +1,155 @@
+import argparse
+import http.server
+import json
+import urllib
+import ssl
+import base64
+
+"""
+Example of setting a network failpoint:
+
+$ curl -X POST localhost:3000
+{"message": "ok"}
+
+$ curl -X GET localhost:3000/set_failpoint/network/1
+{"message": "failpoint set for type: 'network'"}
+
+$ curl -X POST localhost:3000
+curl: (52) Empty reply from server
+
+$ curl -X POST localhost:3000
+{"message": "ok"}
+
+"""
+
+# A new instance of Hander is created for every request, so these have to be global variables
+failpoint_type = None
+remaining_decrypt_fails = 0
+remaining_network_fails = 0
+
+fake_ciphertext = 'a' * 96
+fake_plaintext = 'b' * 96
+
+class HTTPServerWithTLS(http.server.HTTPServer):
+    def __init__(self, server_address, Handler):
+        super().__init__(server_address, Handler)
+        self.socket = ssl.wrap_socket(
+            self.socket,
+            server_side=True,
+            certfile='../x509gen/server.pem',
+            ca_certs='../x509gen/ca.pem',
+            ssl_version=ssl.PROTOCOL_TLS
+        )
+
+
+class Handler(http.server.BaseHTTPRequestHandler):
+    protocol_version = "HTTP/1.1"
+
+    def _send_json(self, data: dict):
+        data_json: bytes = json.dumps(data).encode("utf8")
+        self.send_response(http.HTTPStatus.OK)
+        self.send_header("Content-Type", "application/json")
+        self.send_header("Content-Length", len(data_json))
+        self.end_headers()
+        self.wfile.write(data_json)
+
+    def _send_not_found(self):
+        self.send_response(http.HTTPStatus.NOT_FOUND)
+        msg = "Not found".encode("utf8")
+        self.send_header("Content-Type", "text/plain")
+        self.send_header("Content-Length", len(msg))
+        self.end_headers()
+        self.wfile.write(msg)
+
+    def _decrypt_fail(self):
+        global remaining_decrypt_fails
+        remaining_decrypt_fails -= 1
+        self.send_response(429)
+    
+    def do_GET(self):
```

Comment:
I'm iffy on using GET to modify the server state. Could this go into a PUT/POST handler?